bids/layout/layout.py

```diff
@@ -957,8 +957,11 @@ def get(self, return_type='object', target=None, scope='all',
         # Strip leading periods if extensions were passed
         if 'extension' in filters:
```

Comment:
What if we revert and just make this:

```suggestion
        if 'extension' in filters and 'bids' in self.config:
```

This would mean that users have provided their own config, so they should be expected to correctly pass in extensions, so we don't need to modify them at all.

Reply:
I don't think I quite understand the drop_dot check, but this should also work and would be the least code change